### PR TITLE
{RDBMS} Update Python SDK for PostgreSQL flexible servers to stable version API to 2024-08-01

### PR DESCRIPTION
**Pull request by nasc17**
**Related command**

**Description**
Update CLI to use latest Python SDK version using new package import from https://pypi.org/project/azure-mgmt-postgresqlflexibleservers/1.0.0/

Taking changes from #29992 to Mitigate subprocess security risk.

**Testing Guide**
Integration test run

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by nasc17** [comment]
@evelyn-ys @calvinhzy please review, thank you

File: src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py [hunk 1]
```diff
@@ -375,8 +368,8 @@ def fill_action_template(cmd, database_engine, server, database_name, administra
     if not os.path.exists(action_dir):
         os.makedirs(action_dir)
 
-    process = run_subprocess_get_output("gh secret list --repo {}".format(repository))
-    github_secrets = process.stdout.read().strip().decode('UTF-8')
+    process = run_cmd(["gh", "secret", "list", "--repo", repository], capture_output=True)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py:371]: cc @AllyW for review

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 2]
```diff
@@ -58,6 +58,7 @@ azure-mgmt-monitor==5.0.0
 azure-mgmt-msi==7.0.0
 azure-mgmt-netapp==10.1.0
 azure-mgmt-policyinsights==1.1.0b4
+azure-mgmt-postgresqlflexibleservers==1.0.0
```
Comment (evelyn-ys) [inline src/azure-cli/requirements.py3.Darwin.txt:61]: Will there be another PR remove `azure-mgmt-rdbms` dependency?

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/requirements.py3.Darwin.txt (hunk 2 above)
Comment (nasc17) [reply src/azure-cli/requirements.py3.Darwin.txt:61]: Will MySql and MariaDb teams be adding change to their dependency? Left in since those commands are still dependent.

Comment (evelyn-ys) [reply src/azure-cli/requirements.py3.Darwin.txt:61]: mysql has updated dependency https://github.com/Azure/azure-cli/pull/30123

not sure if mariadb is covered

**Review by AllyW — APPROVED** [review]
